src/FormDataEncoder.ts

```diff
@@ -336,7 +337,15 @@ export class FormDataEncoder {
   async* encode(): AsyncGenerator<Uint8Array, void, undefined> {
     for (const part of this.values()) {
       if (isFile(part)) {
-        yield* part.stream()
+        const stream: any = part.stream()
+        if (stream.getReader instanceof Function) {
+          const reader: any = stream.getReader()
+          let result
+          do {
+            result = await reader.read()
+            if (result.value) yield result.value
+          } while (!result.done)
```

Comment:
I would suggest to replace it with `while(true)` instead

```suggestion
      while (true) {
        const {done, value} = await reader.read()

        if (done) {
          break
        }

        yield value
      }
```